agent/start.go

```diff
@@ -45,10 +45,8 @@ func StartDaemon(ctx context.Context) (*Client, error) {
 	env := os.Environ()
 	env = append(env, "FLY_NO_UPDATE_CHECK=1")
 
-	versionPre := buildinfo.Version().Pre
-
-	if len(versionPre) > 0 {
-		versionNum := versionPre[0].VersionNum
+	if buildinfo.IsDev() {
+		versionNum := buildinfo.ParsedVersion()
 		env = append(env, fmt.Sprintf("FLY_DEV_VERSION_NUM=%d", versionNum))
```

Comment:
I could be misreading this but I think `versionNum` here would be of type `Version`, right?
It's being formatted as `%d`, though - do we want a function that extracts some sort of "build number"-adjacent integer for this case?

Reply:
I wasn't really clear on what problem this code was solving. Clearly it's injecting something in the environment for pre-release builds but I wasn't sure why it did that.

Unfortunately calver-go doesn't expose the micro/build number. We can get a time associated with the components of the year/month/day but that won't take the micro component into account.

Exposing this value upstream would of course be very easy, but I wanted to be clear on what this code was doing before doing that. For this purpose it may be fine to just use a unix timestamp, though of course that'd change on every build so may not be desirable.

Reply:
Maybe. I'm still unclear what purpose putting the version in the environment is supposed to serve. I'm guessing it has something to do with the agent checking whether there's a mismatch, but the variable only appears in 2 places--here and `buildinfo`. It also seems like it may not be intended to contain the entire version, but I'm not sure.

Maybe I'm over-complicating things, but with calver we can fairly easily generate a fake but sensible version if one doesn't exist (I.e. for dev builds.) We can also fairly easily detect that a given generated version is newer/older than whatever binary may have started here. I'm just not clear on what the problem we're solving here is. Maybe the linked PR gives enough context, but not quite for me as a n00b. :)

CC @billyb2

Reply:
yeah I don't really follow this bit of code either. my instinct is to either yank most of this out and simplify it, or just ship what's here and hope for the best. either of those are honestly fine choices.

only thing that gives me pause is the thing I mentioned above: `fmt.Sprintf("FLY_DEV_VERSION_NUM=%d", versionNum)` is using the `%d` formatter to print a Version, which seems wrong, right?

Reply:
You're almost certainly right--I don't know what this specific format string does and I likely copy-pasted something wrong. Fixing that is on my list of things to do, but I too am inclined to yank this code given that I can't find any reference to the environment variable that explains what's going on. And if we yank this code then we're yanking the format string too.

The initial code just dumped a timestamp in that variable. I wanted to be clear on what exactly was going there before I dove into figuring out which data and format string would be best.

Reply:
This code is pretty horrendous looking back :/.

From what I remember, and from reading [an old Discourse post](https://flyio.discourse.team/t/launching-proxy-fails-on-dev-version/1802), this fixes a bug where the fly agent would be different then the version that flyctl used. The fix for this was to set the version number as an environment variable, that fly agent uses. 

The micro component, in this case, is used since fly version numbers look like v0.0.0-1398431841. I think as long as the FLY_DEV_VERSION_NUMBER is a uint, it should be fine. This code might also no longer be relevant, but I'm not sure what changes have been made to flyctl since it was introduced.

(Also wow, I'm getting nostalgia for code I wrote 4 months ago)

Reply:
> yeah I don't really follow this bit of code either. my instinct is to either yank most of this out and simplify it, or just ship what's here and hope for the best. either of those are honestly fine choices, but the first is a bit more work for little reward.
> 
> only thing that gives me pause is the thing I mentioned above: `fmt.Sprintf("FLY_DEV_VERSION_NUM=%d", versionNum)` is using the `%d` formatter to print a Version, which seems wrong, right? Should it be grabbing a unix timestamp from the version, or something?

As for this specifically, this code is correct, it's just a terrible variable name. I'm getting the Version's .Pre member, and then just calling that version number, since that was value that mattered for dev versions at the time

Reply:
OK, thanks for the context. I went ahead and removed the environment variable code for now, particularly since I probably wasn't doing the right thing with it in version.go anyway. If we need it back we can re-add it.

Right now my expectation is that releases will have versions like 2023.07.01.1 while dev versions will have versions like 2023.07.01.1234567890 (E.g. followed by the build/run timestamp.) So they'll always be different and we shouldn't need to worry about outdated agents.

Thanks again.